### ISSUE TITLE
How to train to get the model with mAP of 28.54

### ISSUE THREAD

**Issue by dreamPoet** [fredzzhang/spatio-attentive-graphs#30]
Sorry to bother you but would you mind giving me more tips about how to achieve the mAP of 28.54? I download the fasterrcnn_resnet50_fpn_hicodet_e13.pt but do not know how to use it. Shall I train the network with the detections produced with this .pt file? With python preprocessing.py --partition test2015 ckpt_path='path to .pt'? (Did u forget an h after .pt?)

Thx.

**Comment by dreamPoet** [comment]
If I need to download the detector from DRG github directly, which one I should use?

**Comment by fredzzhang** [comment]
Hi, @dreamPoet 

Thanks for taking interest in our work.

You just need to follow the instructions on the front page. I'll list them here again

1. Download the HICO-DET dataset
2. Run a Faster R-CNN pre-trained on MS COCO to generate detections
```bash
cd /path/to/spatio-attentive-graphs/hicodet/detections
python preprocessing.py --partition train2015
```
Note that if you ONLY want to reproduce the results with `28.54` mAP, you do NOT need to generate test set detections with pre-trained detector. Instead, download the fine-tuned detections.
3. Download fine-tuned detections
```bash
cd /path/to/spatio-attentive-graphs/download
bash download_finetuned_detections.sh
```
4. Train the model. The checkpoints will be saved under `checkpoints` in the current directory
```bash
cd /path/to/spatio-attentive-graphs
CUDA_VISIBLE_DEVICES=0 python main.py &>log &
```
5. Test the model. Based on the instructions, the model at epoch `11` is picked.
```bash
cd /path/to/spatio-attentive-graphs
CUDA_VISIBLE_DEVICES=0 python test.py --model-path checkpoints/ckpt_103499_11.pt
```

Mostly likely you won't get exactly the same number from your trained model (like in #29), but you should be able to get something similar to `28.54` mAP. 

Let me know if you have other problems.

Cheers,
Fred

**Comment by dreamPoet** [comment]
Thank u for your reply. But is there a way to produce these detections with DRG detector by myself?

**Comment by fredzzhang** [comment]
I emailed the authors of DRG. They said that they have deleted the weights of their fine-tuned detector. So the detections they provided are all we have access to.

**Comment by dreamPoet** [comment]
So is it possible to get the detections via training a DRG model by myself? 

BTW, I am trying to train via `CUDA_VISIBLE_DEVICES=0 python main.py &>log &` without changing any hyper-parameters. In your paper you said only 48 hours needed to finish the training, but I did not get any saved checkpoints til now... how long does it take to finish one epoch?

**Comment by fredzzhang** [comment]
> So is it possible to get the detections via training a DRG model by myself?

That's beyond the scope of this repo. You'll have to look into the DRG repo and contact the authors.

> how long does it take to finish one epoch?

It depends on your hardware. For a GeForce GTX TITAN X device, it takes about 3.5 hours for finish one epoch. More powerful hardwares will be able to train faster

**Comment by dreamPoet** [comment]
I see. Then is there a way to set CUDA_VISIBLE_DEVICES=0 (,1,2,3) wiith srun at the same time...? Especially for the multi gpu version..

If I use a larger batchsize, for example, 8*4, I shall set the learning-rate as 8*0.001?

**Comment by fredzzhang** [comment]
The code for multi-gpu training, as mentioned in the instructions, has not been fully tested. The suggested learning rate is only a rule of thumb the community uses. So there is no guarantee that it will reproduce the same performance. You might want to play with those hyper-parameters to find the optimal setting.

**Comment by fredzzhang** [comment]
Since there hasn't been any recent posts, I assume I can close this now.
